ESLint fails with `no-invalid-this` on http_plugin_unit, when including Shaka repo in my project folder

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**:
Yes

**What version of Shaka Player are you using?**:
Latest from master, but version does not seem to matter, this is a problem with the build

**What browser and OS are you using?**:
MacOS, not a problem in the browser

**What did you do?**
I have shaka-player repo as a git submodule in my own project's repository. (However, I see this problem whether shaka-player is an actual git submodule or not. That is, if I just copy-paste the shaka-player repo to somewhere in my project folder, I still see this problem.)

**What did you expect to happen?**
Running `./build/all.py` should build as expected.

**What actually happened?**
ESLint fails with `no-invalid-this` in http_plugin_unit:

```
[INFO] Generating Closure dependencies...
[INFO] Running eslint...

/Users/chrisf/Documents/repo/vstb/html5/submodules/shaka-player/test/net/http_plugin_unit.js
  304:9  error  Unexpected 'this'  no-invalid-this
  306:9  error  Unexpected 'this'  no-invalid-this
  309:9  error  Unexpected 'this'  no-invalid-this
  311:9  error  Unexpected 'this'  no-invalid-this

✖ 4 problems (4 errors, 0 warnings)
```

Shaka's eslintrc configuration includes `no-invalid-this`. Based on the description here:

https://eslint.org/docs/rules/no-invalid-this

...it seems like the offending usage of `this` should fail all the time. Have I misunderstood?

We also use ESLint in our project. I wonder if somehow configuration gets polluted in these scenarios? I suppose this could be an ESLint bug, but I can work around it by changing `newXHRMock` to `NewXHRMock`, so I will file a PR.

--- a/test/net/http_plugin_unit.js
+++ b/test/net/http_plugin_unit.js
@@ -300,7 +300,7 @@ function httpPluginTests(usingFetch) {
       // actually insert a call to abort in the middle.
       // Instead, install a very elementary mock.
       /** @constructor */
-      let newXHRMock = function() {
+      let NewXHRMock = function() {
         this.abort = shaka.test.Util.spyFunc(jasmine.createSpy('abort'));
 
         this.open = shaka.test.Util.spyFunc(jasmine.createSpy('open'));
@@ -318,7 +318,7 @@ function httpPluginTests(usingFetch) {
           }.bind(this));
         };
       };
-      shaka.net.HttpXHRPlugin['Xhr_'] = newXHRMock;
+      shaka.net.HttpXHRPlugin['Xhr_'] = NewXHRMock;
 
       let request = shaka.net.NetworkingEngine.makeRequest(
           ['https://foo.bar/'], retryParameters);

